internal/file/file_manager_service.go

```diff
@@ -23,15 +24,30 @@ import (
 	backoffHelpers "github.com/nginx/agent/v3/internal/backoff"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6@v6.8.1 -generate
+//counterfeiter:generate . fileOperator
+
+type (
+	fileOperator interface {
```

Comment:
do we need to check for interface compliance here?

Reply:
Interface compliance is checked in file_operator.go https://github.com/nginx/agent/blob/b87eeb1d3d14c3a2e4ad076e18a8a518a1b55b6d/internal/file/file_operator.go#L22